app/src/main/java/com/yourssu/unscramble/presentation/play/PlayViewModel.kt

```diff
@@ -1,27 +1,85 @@
+    val questionWord: StateFlow<String> = _questionWord.asStateFlow()
 
-    private val _currentScore: MutableStateFlow<String> = MutableStateFlow("0")
-    val currentScore: MutableStateFlow<String> = _currentScore
+    private val _currentScore: MutableStateFlow<Int> = MutableStateFlow(0)
+    val currentScore: StateFlow<Int> = _currentScore.asStateFlow()
 
-    private val _solvedProblem: MutableStateFlow<String> = MutableStateFlow("0")
-    val solvedProblem: MutableStateFlow<String> = _solvedProblem
+    private val _solvedProblem: MutableStateFlow<Int> = MutableStateFlow(1)
+    val solvedProblem: StateFlow<Int> = _solvedProblem.asStateFlow()
 
     private val _timerHour: MutableStateFlow<String> = MutableStateFlow("00")
-    val timerHour: MutableStateFlow<String> = _timerHour
+    val timerHour: StateFlow<String> = _timerHour.asStateFlow()
 
     private val _timerMinute: MutableStateFlow<String> = MutableStateFlow("00")
-    val timerMinute: MutableStateFlow<String> = _timerMinute
+    val timerMinute: StateFlow<String> = _timerMinute.asStateFlow()
 
-    private val _submitButtonText: MutableStateFlow<String> = MutableStateFlow("SUBMIT")
-    val submitButtonText: MutableStateFlow<String> = _submitButtonText
+    // 유효성 판단에 따른 버튼 활성화 관련 코드
+    // 사용자가 입력한 답안
+    private val _inputAnswer: MutableStateFlow<String> = MutableStateFlow("")
 
-    private val _skipButtonText: MutableStateFlow<String> = MutableStateFlow("SKIP")
-    val skipButtonText: MutableStateFlow<String> = _skipButtonText
+    // 입력 답안 유효성 검사
+    private val answerIsValid =
+        _inputAnswer.map { input ->
+            input.isNotBlank() && REGEX.matcher(input).matches()
+        }
+
+    // textField
+    private val _textFieldIsNegative: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    val textFieldIsNegative: StateFlow<Boolean> = _textFieldIsNegative.asStateFlow()
+
+    // submit 버튼 활성화
+    private val _submitBtnIsDisable: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    val submitBtnIsDisable: StateFlow<Boolean> = _submitBtnIsDisable.asStateFlow()
+
+    private val _navigateToEnd: MutableSharedFlow<Boolean> = MutableSharedFlow()
+    val navigateToEnd: SharedFlow<Boolean> = _navigateToEnd.asSharedFlow()
+
+    fun onPlayButtonClick() {
+        if (_solvedProblem.value == 9) {
+            viewModelScope.launch {
+                _navigateToEnd.emit(true)
+            }
+        } else {
+            updatePlayView()
+            checkValid()
+        }
+    }
+
+    fun checkValid() {
+        viewModelScope.launch {
+            answerIsValid.collect { result ->
+                _textFieldIsNegative.value = !result
+                _submitBtnIsDisable.value = !result
+            }
+        }
+    }
+
+    fun setInputAnswer(answer: String) {
+        _inputAnswer.value = answer
+    }
+
+    private fun updatePlayView() {
+        _solvedProblem.value = (solvedProblem.value + 1)
```

Comment:
```suggestion
        _solvedProblem.update { it + 1 }
```
[update](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/update.html) 함수를 사용해서 스레드 안전하게 업데이트하는 방법도 있어요!

https://medium.com/geekculture/atomic-updates-with-mutablestateflow-dc0331724405